.github/workflows/cd.yml

```diff
@@ -356,58 +361,69 @@ jobs:
-              echo 'publish-docs=true'
-            } >> "$GITHUB_OUTPUT"
-            exit 0
-          fi
-
-          # Define environments matrix,
-          # and deploy docs only if we are publishing to prod
-          if [ "${ENVIRONMENT}" == 'dev' ]; then
-            {
-              echo 'environments=["dev"]'
-              echo 'publish-docs=false'
-            } >> "$GITHUB_OUTPUT"
-          elif [ "${ENVIRONMENT}" == 'ops' ]; then
-            {
-              echo 'environments=["ops"]'
-              echo 'publish-docs=false'
-            } >> "$GITHUB_OUTPUT"
-
-          elif [ "${ENVIRONMENT}" == 'prod' ]; then
-            {
-              echo 'environments=["dev", "ops", "prod"]'
-              echo 'publish-docs=true'
-            } >> "$GITHUB_OUTPUT"
-
-          elif [ -z "${ENVIRONMENT}" ] || [ "${ENVIRONMENT}" == 'none' ]; then
-            {
-              echo 'environments=[]'
-              echo 'publish-docs=false'
-            } >> "$GITHUB_OUTPUT"
-
-          else
-            echo "Invalid environment: ${ENVIRONMENT}"
-            exit 1
-          fi
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            // Helper to get input/env values
+            const getInput = (name, fallback = '') => {
+              return process.env[name] !== undefined ? process.env[name] : fallback;
+            };
+
+            const ENVIRONMENT = getInput('ENVIRONMENT');
+            const DOCS_ONLY = getInput('DOCS_ONLY');
+            const HAS_BACKEND = getInput('HAS_BACKEND');
+
+            // Platforms matrix
+            let platforms;
+            if (HAS_BACKEND === 'true') {
+              platforms = ['linux', 'darwin', 'windows', 'any'];
+            } else {
+              platforms = ['any'];
+            }
+            core.setOutput('platforms', JSON.stringify(platforms));
+
+            // Docs only: skip plugin deployment
+            if (DOCS_ONLY === 'true') {
+              core.setOutput('environments', JSON.stringify([]));
+              core.setOutput('publish-docs', 'true');
+              return;
+            }
+
+            // Special case to skip deployment
+            if (!ENVIRONMENT || ENVIRONMENT === 'none') {
+              core.setOutput('environments', JSON.stringify([]));
+              core.setOutput('publish-docs', 'false');
+              return;
+            }
+
+            // Parse and filter environments
+            const allowedEnvironments = ['dev', 'ops', 'prod'];
+            let environments = [];
+            environments = ENVIRONMENT.split(',')
+              .map(e => e.trim())
+              .filter(e => allowedEnvironments.includes(e));
+
+            // Special case: 'prod' means all environments
+            if (environments.includes('prod')) {
+              environments = allowedEnvironments;
+            }
+
+            // Remove duplicates and sort
```

Comment:
I think the "sort" part is missing, you could add `.sort()` to the Array.

`environments = Array.from(new Set(environments)).sort();`